### DISCUSSION TITLE
Official skript addon example template

### DISCUSSION THREAD

**Discussion by AyhamAl-Ali** [SkriptLang/Skript#4654] [General]
# Skript Addons
Many developers have struggled when making their first skript addon due to the lack of documentation (currently) and tried to find some simple skript addons on the internet to base off of them and that's annoying.

Therefore, an official skript addon template would be very helpful and a very good way to make the process easier for making your new skript addon instead of starting from scratch and help you finding the methods and the structures that you will probably need.

So, in this discussion share and discuss what you want to see in a skript addon example template to help you with your first/next skript addon.

As a beginning, the example template should have the following:
- Creating a **Condition**
- Creating a **Section**
- Creating a **SectionEffect**
- Creating an **Effect**
- Creating a **Type/ClassInfo**
- Creating an **Event** (Bukkit/Skript)
- Creating an **Expression** (Simple/Property/Wrapper)
- Creating **Enum names**
- Registering the addon

_Your turn now!_

**Comment by mor-uto** [comment]
where is the thing??

**Reply by Fusezion** [reply]
Lime was working on this a while ago, I haven't seen any further examples for this but skunity does have an older guide for this which for the most part is up to date

**Reply by mor-uto** [reply]
ok, ill be waiting for it 👍

**Reply by mor-uto** [reply]
its never coming ;(

**Reply by Fusezion** [reply]
I mean it takes time but the old guide on skunity is still relevant.
